File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/AcmePizza/AcmePizza_CSharp/PizzaOrder.cs
//--------------------------------------------------------------------------
// 
//  Copyright (c) Microsoft Corporation.  All rights reserved. 
// 
//  File: PizzaOrder.cs
//
//--------------------------------------------------------------------------

namespace AcmePizza
{
    public enum PizzaToppings
    {
        Mushrooms,
        Olives,
        Sausage,
        Pepperoni,
        Cheese,
        Onions,
        BellPeppers,
        Bacon,
        Chicken,
        Artichokes,
    }

    public enum OrderSource
    {
        Phone,
        Internet,
        WalkIn,
        Fax,
    }

    public struct PizzaOrder
    {
        public OrderSource Source {get;set; }
        public string PhoneNumber { get; set; }
        public int Size { get; set; }
        public bool IsDelivery { get; set; }
        public PizzaToppings[] Toppings { get; set; }
    }
}

File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/AcmePizza/AcmePizza_VisualBasic/PizzaOrder.vb
'--------------------------------------------------------------------------
' 
'  Copyright (c) Microsoft Corporation.  All rights reserved. 
' 
'  File: PizzaOrder.vb
'
'--------------------------------------------------------------------------

Namespace AcmePizza
	Public Enum PizzaToppings
		Mushrooms
		Olives
		Sausage
		Pepperoni
		Cheese
		Onions
		BellPeppers
		Bacon
		Chicken
		Artichokes
	End Enum

	Public Enum OrderSource
		Phone
		Internet
		WalkIn
		Fax
	End Enum

	Public Structure PizzaOrder
		Public Property Source() As OrderSource
		Public Property PhoneNumber() As String
		Public Property Size() As Integer
		Public Property IsDelivery() As Boolean
		Public Property Toppings() As PizzaToppings()
	End Structure
End Namespace
